### PR TITLE
Improve GraalVM script engines interoperability.

### PR DESCRIPTION
**Pull request by sdedic**
There are a few types of changes:
- the exceptions thrown are better reported: the original exception is attached as the `cause` of the `RuntimeException` (actually `PolyglotException`) thrown from eval/invoke methods.
- better (un)wrapping of `org.graalvm.polyglot.Value` from/to POJOs
- the `ScriptEngine` returned by `Scripting.createManager()` will support `ScriptContext.ENGINE` scope as the top-level language scope.
- global attributes are now supported, even before the first `eval` or other `Context`-creating method on the Engine is called.

I also somewhat checked the testu results agains GraalVM 20.1 (now travis runs 19.3.1)

### CODE REVIEW

File: ide/libs.graalsdk/src/org/netbeans/libs/graalsdk/impl/GraalEngine.java [hunk 1]
```diff
@@ -55,20 +57,32 @@ public Object eval(String src, ScriptContext arg1) throws ScriptException {
...
+        public Value run();
+    }
+    
+    private Value handleException(ScriptAction r) throws ScriptException {
         try {
-            return ((GraalContext) arg1).ctx().eval(id(), src);
+            return r.run();
         } catch (PolyglotException e) {
+            if (e.isHostException()) {
+                e.initCause(e.asHostException());
```
Comment (JaroslavTulach) [inline ide/libs.graalsdk/src/org/netbeans/libs/graalsdk/impl/GraalEngine.java:70]: OK.

**Comment by neilcsmith-net** [comment]
Is the Travis failure in the GraalVM tests a concern or coincidence?

**Comment by sdedic** [comment]
> Is the Travis failure in the GraalVM tests a concern or coincidence?

Concern.  Apologies, I've didn't run full tests.

Actually ... it's very interesting. @JaroslavTulach could have more deeper insight in this matter. The failure is actually caused by  **classloader mismatch**. 

`GraalJSTest` - the testcase actually run by `junit` ant task is loaded by AppClassLoader, while hamcrest is being loaded by ModuleSystem$SystemClassLoader. There are two inconsistencies:
- NbModuleSuite only loads `org.junit.**` from AppClassLoader, not `org.hamcrest.**`
- the `lib.junit4` module does not list `org.hamcrest.**` as public packages, although it contains the relevant classes.

**Comment by JaroslavTulach** [comment]
>     * NbModuleSuite only loads `org.junit.**` from AppClassLoader, not `org.hamcrest.**`

cc43414 seems to be the correct fix.

**Comment by JaroslavTulach** [comment]
Sváťa has gone for a vacation. @neilcsmith-net if it is OK to merge, feel free to do so. I am convinced the change prepared by Sváťa improves the exception handling quite significantly.

**Comment by sdedic** [comment]
Yes, actually waiting for re-approval after the changes. Sorry, I was out of Internet connection yesterday

**Comment by neilcsmith-net** [comment]
@sdedic go ahead from my perspective. Your choice whether to squash or not first.  Whatever makes sense.

**Pull request merged**